File: src/ProjectR/content/scripts/afflictions/Passives/Minion.lua
function GetName()
  return "Minion"
end

function GetType()
  return Passive
end

function GetHookPoints()
  return {  }
end

function OnAttachment(char)
  local stats = GetStats(char)
  local hpStat = stats:GetSingleStat(HP)
  local baseHP = hpStat:Get(Base)
  baseHP = baseHP * 0.3
  hpStat:Set(Base, baseHP)

  for i = AD, MR do
    local stat = stats:GetSingleStat(i)
    local baseStat = stat:Get(Base)
    baseStat = baseStat * 0.5
    stat:Set(Base, baseStat)
  end

  char:Heal(char:GetMaxHP())
end

function OnRemoval(char)
  local stats = GetStats(char)
  local hpStat = stats:GetSingleStat(HP)
  local baseHP = hpStat:Get(Base)
  baseHP = baseHP / 0.3
  hpStat:Set(Base, baseHP)

  for i = AD, MR do
    local stat = stats:GetSingleStat(i)
    local baseStat = stat:Get(Base)
    baseStat = baseStat / 0.5
    stat:Set(Base, baseStat)
  end
end

function Trigger(hookPoint, char)
end

File: src/ProjectR.Scripting/content/scripts/afflictions/Passives/Minion.cs
using ProjectR.Interfaces.Model;
using ProjectR.Interfaces.Model.Stats;

namespace ProjectR.Scripting.Afflictions
{
    public class Minion : Affliction
    {
        public override string Name { get { return "Minion"; } }

        public override AfflictionType Type { get { return AfflictionType.Passive; } }

        protected override HookPoint[] HookPoints
        {
            get
            {
                return new HookPoint[]
                {
                    
                };
            }
        }

        protected override void OnAttachment(ICharacter character)
        {
            var stats = character.Stats;
            stats[BaseStat.HP][StatType.Base] *= .3;

            for (var i = BaseStat.AD; i <= BaseStat.MR; ++i)
            {
                stats[i][StatType.Base] *= .5;
            }

            character.Heal(stats.GetTotalStat(BaseStat.HP));
        }

        protected override void OnRemoval(ICharacter character)
        {
            var stats = character.Stats;
            stats[BaseStat.HP][StatType.Base] /= .3;

            for (var i = BaseStat.AD; i <= BaseStat.MR; ++i)
            {
                stats[i][StatType.Base] /= .5;
            }

            character.Heal(stats.GetTotalStat(BaseStat.HP));
        }
    }
}